src/tools/connection_patcher/Program.cpp

```diff
@@ -32,6 +32,9 @@
 
 #if PLATFORM == PLATFORM_WINDOWS
 #include <Shlobj.h>
+#elif PLATFORM == PLATFORM_UNIX
```

Comment:
As I can see in the original code OS X was fully supported.

Answering my own question, yes it's false (PLATFORM_APPLE is used for OS X: https://github.com/TrinityCore/TrinityCore/blob/6.x/src/server/shared/CompilerDefs.h)